mediacapture-streams/MediaStreamTrack-applyConstraints.https.html

```diff
@@ -8,51 +8,46 @@
-            t.unreached_func('Accepted invalid groupID'),
-            t.step_func(e => {
-              assert_equals(e.name, 'OverconstrainedError');
-              assert_equals(e.constraint, 'groupId');
-            }));
-      }));
+  promise_test(async t => {
+    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+    const [track] = stream.getVideoTracks();
+    t.add_cleanup(() => track.stop());
+    try {
+      await track.applyConstraints({ groupId: { exact: "INVALID" } });
+      t.unreached_func('Accepted invalid groupID')();
+    } catch (e) {
+      assert_equals(e.name, 'OverconstrainedError');
+      assert_equals(e.constraint, 'groupId');
+    }
   }, 'applyConstraints rejects invalid groupID');
 
-  promise_test(t => {
-    return navigator.mediaDevices.getUserMedia({ video: true })
-      .then(t.step_func(stream => {
-        var track = stream.getVideoTracks()[0];
-        var groupId = track.getSettings().groupId;
-        return track.applyConstraints({ groupId: "INVALID" }).then(
-          t.step_func(() => {
-            assert_equals(track.getSettings().groupId, groupId);
-          }));
-      }));
+  promise_test(async t => {
+    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+    const [track] = stream.getVideoTracks();
+    t.add_cleanup(() => track.stop());
+    const groupId = track.getSettings().groupId;
+    await track.applyConstraints({ groupId: "INVALID" });
+    assert_equals(track.getSettings().groupId, groupId);
   }, 'applyConstraints accepts invalid ideal groupID, does not change setting');
 
-  promise_test(t => {
-    return navigator.mediaDevices.getUserMedia({ video: true })
-      .then(t.step_func(stream => {
-        var track = stream.getVideoTracks()[0];
-        var groupId = track.getSettings().groupId;
-        return navigator.mediaDevices.enumerateDevices().then(devices => {
-          var anotherDevice = devices.find(device => {
-            return device.kind == "videoinput" && device.groupId != groupId;
-          });
-          if (anotherDevice !== undefined) {
-            return track.applyConstraints(
-              { groupId: { exact: anotherDevice.groupId } }).then(
-                t.unreached_func(),
-                t.step_func(e => {
-                  assert_equals(e.name, 'OverconstrainedError');
-                  assert_equals(e.constraint, 'groupId');
-                }));
-          }
-        });
-      }));
+  promise_test(async t => {
+    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+    const [track] = stream.getVideoTracks();
+    t.add_cleanup(() => track.stop());
+    const groupId = track.getSettings().groupId;
+    const devices = await navigator.mediaDevices.enumerateDevices();
+    const anotherDevice = devices.find(device => {
+      return device.kind == "videoinput" && device.groupId != groupId;
+    });
+    if (anotherDevice !== undefined) {
```

Comment:
Shouldn't we assert instead of skipping test here? Seems rare for a test system to have one device but not two (audio and video)...

Reply:
I will fix it in the next PR.